### ISSUE TITLE
`await` hangs endlessly when calling a function directly in conjunction with the constructor

### ISSUE THREAD

**Issue by MikeSchulze** [godotengine/godot#80292]
### Godot version

v4.0.4.stable.official [fc0b241c9]

### System information

Windows 10

### Issue description

I run today into this issue, i build a plugin thats supports all Godot 4 versions and is broken for all Godot 4.0.1 until 4.0.4.
Running on Godot 4.1.x works fine.

In my case i have defined an assertion to test executed code `await assert_error(<callable>).is_success()` and needs to await the code is executed and finally test if is success.

### Steps to reproduce

Simplified example code is:

```gdscript
class X:
	func wait():
		await Engine.get_main_loop().process_frame

func test():
	await X.new().wait()` -> broken it hangs but works since Godot 4.1

        # works on all Godot 4 versions
	var x := X.new()
	await x.wait()

```

### Minimal reproduction project

[IssuesAwaitFunc.zip](https://github.com/godotengine/godot/files/12266672/IssuesAwaitFunc.zip)

**Comment by dalexeev** [comment]
I think this is a bug with temporaries, which was fixed by #74101 for 4.1. I'm not sure about the status of 4.0 branch, if it's possible to cherry-pick the fix, if it would be considered safe enough.

CC @RandomShaper @akien-mga @YuriSizov

**Comment by MikeSchulze** [comment]
i can live with the workaround, but i need to document this in my plugin

**Comment by YuriSizov** [comment]
Could you confirm that this is no longer an issue with 4.1/4.1.1, so we can close this? The fix is indeed not suitable for a 4.0.x patch release, so we can't address this for the 4.0 branch.

**Comment by MikeSchulze** [comment]
it is fixed on 4.1 and 4.1.1 yes
